### DISCUSSION TITLE
[docs] [Windows]How to make the title bar transparent?

### DISCUSSION THREAD

**Discussion by thep0y** [tauri-apps/tauri#8387] [Q&A] [answered]
The `TitleBarStyle` can only be used on `macOS`.

https://github.com/tauri-apps/tauri/blob/803c3a794d96c1f0b3361ca98f56e8bcf5038ede/core/tauri-utils/src/lib.rs#L171-L185

I would like the title bar's color to match the background color, but I haven't found any interface to set the title bar color or make it transparent on the `Windows` platform.

I have set the window to be transparent, but this does not affect the title bar:

```rust
let window = builder.transparent(true).build().unwrap();
```

<img width="648" alt="image" src="https://github.com/tauri-apps/tauri/assets/51874567/8b3ad7ca-cec6-41e5-aa2e-9a7c892a4c25">

I am not sure if this is a documentation issue or a new feature.

**Comment by amrbashir** [comment, marked as answer]
On Windows, you can't really have a transparent native titlebar, instead, most apps will hide the native titlebar (`.decorations(false)`) and create a custom one using HTML/CSS/JS.

A few references:
https://tauri.app/v1/guides/features/window-customization/
https://github.com/agmmnn/tauri-controls

**Reply by thep0y** [reply]
I am currently using the method you mentioned, but I am unable to trigger the hover event for the maximize button.

Like VSCode:

<img width="426" alt="image" src="https://github.com/tauri-apps/tauri/assets/51874567/a38bb0db-3394-4be2-a972-bad79734635b">

**Reply by emipasca** [reply]
did you find a solution?

**Reply by thep0y** [reply]
> did you find a solution?

Nope.

Tauri's compatibility with native interfaces across various platforms isn't quite there yet, and I'm unsure whether they have intentions to refine this in the future.

**Reply by timwehrle** [reply]
> > did you find a solution?
> 
> Nope.
> 
> Tauri's compatibility with native interfaces across various platforms isn't quite there yet, and I'm unsure whether they have intentions to refine this in the future.

Absolutely, but can you show your code to make it easier to see where you encounter your problems?

**Reply by FabianLars** [reply]
> and I'm unsure whether they have intentions to refine this in the future.

Our plans don't really matter if the webviews prevent us from doing stuff lol. For the snap overlay we have an idea that is based on the new borderless resizing implementation but that may still take a while. In the meantime there's https://github.com/clearlysid/tauri-plugin-decorum which can open the snap overlay by emitting the Win+Z key combination.
